src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -373,12 +373,20 @@ bool IsInHtmlAttributeName()
             //
             // `<p |csharpattr="Variable">` - true
             //
+            // or
+            //
+            // `<p csharpattr="Variable"`
+            // `  |csharpattr="Variable">` - depending on dontFormatIfAttributeIsInNewLine
+            //
             // Because we map attributes, so rename and FAR works, there could be C# mapping for them,
             // but only if they're actually bound attributes
 
-            return owner.AncestorsAndSelf().Any(
-                n => n is MarkupTagHelperAttributeSyntax { TagHelperAttributeInfo: { Bound: true } } or
-                          MarkupTagHelperDirectiveAttributeSyntax { TagHelperAttributeInfo: { Bound: true } });
+            // MarkupTagHelperAttributeSyntax { TagHelperAttributeInfo: { Bound: true } } or
+            var isInHtmlAttributeName = owner.AncestorsAndSelf().Any(
+                n => n is MarkupTagHelperAttributeSyntax { TagHelperAttributeInfo: { Bound: true } } markupTagHelperAttribute
+                    ? dontFormatIfAttributeIsInNewLine || markupTagHelperAttribute.NamePrefix.LiteralTokens[0].Kind != SyntaxKind.NewLine
```

Comment:
> markupTagHelperAttribute.NamePrefix.LiteralTokens[0].Kind != SyntaxKind.NewLine

If this logic does end up staying, I'm not sure using the syntax tree for this is the best. Should be able to get the first non-whitespace character of the line with the following code:

```
sourceText.GetLineAndOffset(absoluteIndex, out var lineIndex, out var offset);
var line = sourceText.Lines[lineIndex];
var lineStart = line.Start + line.GetFirstNonWhitespaceOffset();
```

Comparing that `lineStart` to the start of the actual attribute name seems to me like it would be better, in case the tree structure changes. eg, what if there is whitespace before the newline, is that now part of the `NamePrefix` too?